pkg/gzip/gzip.go

```diff
@@ -55,6 +55,16 @@ func ExtractTarGz(owner string, fp io.Reader, dest string) (err error) {
 			uid, gid = ownerUID, ownerGID
 		}
 
+		parent := filepath.Dir(fpath)
+		if err := os.MkdirAll(parent, 0755); err != nil {
```

Comment:
You'll almost certainly want to use ` util.MkdirChownAll(dest, ownerUID, ownerGID, 0755)`